transformer_engine/common/__init__.py

```diff
@@ -7,12 +7,10 @@
 
 def get_te_path():
     """Find TE path using pip"""
-
-    import os
-
-    te_info = (
-        os.popen("pip show transformer_engine").read().replace("\n", ":").split(":")
-    )
+    import subprocess
+    command = "pip show transformer_engine"
+    result = subprocess.run(command, shell=True, text=True, stdout=subprocess.PIPE, stderr=subprocess.PIPE)
```

Comment:
[`shell=True` is not recommended for safety reasons](https://docs.python.org/3/library/subprocess.html#security-considerations).